t/129-ssl-socket.t

```diff
@@ -2618,3 +2676,410 @@ qr/\[error\] .* ngx.socket sslhandshake: expecting 1 ~ 5 arguments \(including t
 --- no_error_log
 [alert]
 --- timeout: 5
+
+
+
+=== TEST 33: setsslcert, too many arguments
+--- config
+    resolver $TEST_NGINX_RESOLVER ipv6=off;
+    location /t {
+        content_by_lua_block {
+            local sock = ngx.socket.tcp()
+            sock:settimeout(5000)
+
+            local ok, err = sock:connect("openresty.org", 443)
+            if not ok then
+                ngx.say("failed to connect: ", err)
+                return
+            end
+
+            ngx.say("connected: ", ok)
+
+            local ok, err = sock.setsslcert()
+        }
+    }
+
+--- request
+GET /t
+
+--- ignore_response
+--- error_log eval
+qr/\[error\] .* ngx.socket setsslcert: expecting 1 ~ 4 arguments \(including the object\), but seen 0/
+--- timeout: 5
+
+
+
+=== TEST 34: setsslcert should return error on closed connection
+--- config
+    server_tokens off;
+    resolver $TEST_NGINX_RESOLVER ipv6=off;
+    lua_ssl_trusted_certificate ../html/ca.crt;
+
+    location /t {
+        content_by_lua_block {
+            function read_file(file)
+                local f = io.open(file, "rb")
+                local content = f:read("*all")
+                f:close()
+                return content
+            end
+
+            local sock = ngx.socket.tcp()
+            sock:settimeout(3000)
+
+            local cert = read_file("$TEST_NGINX_HTML_DIR/client.crt")
+            local key = read_file("$TEST_NGINX_HTML_DIR/client.unsecure.key")
+
+            local ok, err = sock:setsslcert(cert, key)
```

Comment:
@agentzh I agree ` Loading the raw certificates upon every SSL cosocket is quite expensive and the client-side certificates tend to be reused over and over again.` So how about the new directive `lua_ssl_client_key` and `lua_ssl_client_certificate` to feed  SSL ctx object to it directly? Also can we provide this Lua API to allow user to set different certificates on the fly?

So sorry to long time  reply:(